### PR TITLE
STOR-2017: Bump all deps to 4.18

### PR DESCRIPTION
**Pull request by dfajmon**
Bump all deps to 4.18

Notes:
- Fixed typo in variable `userAgent`
- Updated calls for functions coming from library-go

### CODE REVIEW

**Comment by dfajmon** [comment]
most of the check fail on, seems like prometheus lib update broke it?
```
Operator unavailable (UpdatingPrometheusFailed): UpdatingPrometheus: Prometheus "openshift-monitoring/k8s": NoPodReady: shard 0: pod prometheus-k8s-0: 0/6 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/6 nodes are available: 6 Preemption is not helpful for scheduling.
shard 0: pod prometheus-k8s-1: 0/6 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/6 nodes are available: 6 Preemption is not helpful for scheduling.  Operator unavailable (UpdatingPrometheusFailed): UpdatingPrometheus: Prometheus "openshift-monitoring/k8s": NoPodReady: shard 0: pod prometheus-k8s-0: 0/6 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/6 nodes are available: 6 Preemption is not helpful for scheduling.
shard 0: pod prometheus-k8s-1: 0/6 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/6 nodes are available: 6 Preemption is not helpful for scheduling.}
```

**Comment by dobsonj** [comment]
The hypershift job is failing because storage is not available:
```
    util.go:345: ClusterVersionSucceeding=False: ClusterOperatorNotAvailable(Cluster operator storage is not available)
    util.go:345: ClusterVersionProgressing=True: ClusterOperatorNotAvailable(Unable to apply 4.18.0-0.ci.test-2024-10-24-093800-ci-op-si76t6kd-latest: the cluster operator storage is not available)
```
That's highly suspicious, I don't see this failure for other PR's, but I don't yet see what caused this in the [artifacts](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_csi-operator/301/pull-ci-openshift-csi-operator-master-hypershift-aws-e2e-external/1849418480936292352/artifacts/hypershift-aws-e2e-external/dump-management-cluster/artifacts/).

The other failures look unrelated. e2e-azurestack-csi permafails, security is a [known issue](https://github.com/openshift/csi-operator/pull/295), smb-operator-e2e-extended passed all the tests but failed to gather metrics at the end.

**Comment by dobsonj** [comment]
CVO thinks the DaemonSet never fully deployed:
`./namespaces/clusters-e4fe0048ff5b8aa9d38d-mgmt/core/pods/logs/cluster-version-operator-789cf746d8-p9hxw-cluster-version-operator.log:23328:I1024 12:21:36.578041       1 sync_worker.go:1196] Update error 350 of 620: ClusterOperatorNotAvailable Cluster operator storage is not available (*errors.errorString: cluster operator storage is Available=False: AWSEBSCSIDriverOperatorCR_AWSEBSDriverNodeServiceController_Deploying: AWSEBSCSIDriverOperatorCRAvailable: AWSEBSDriverNodeServiceControllerAvailable: Waiting for the DaemonSet to deploy the CSI Node Service)`

But it looks like it started okay in `hostedcluster-e4fe0048ff5b8aa9d38d-mgmt/namespaces/openshift-cluster-csi-drivers/apps/daemonsets.yaml`:
```
  status:
    currentNumberScheduled: 3
    desiredNumberScheduled: 3
    numberAvailable: 3
    numberMisscheduled: 0
    numberReady: 3
    observedGeneration: 1
    updatedNumberScheduled: 3
```

**Comment by dfajmon** [comment]
ci/prow/hypershift-aws-e2e-external is failing probably due to undated libraries and old kubernetes package
waiting for this to be merged https://github.com/openshift/hypershift/pull/4927

File: pkg/clients/builder.go [hunk 1]
```diff
@@ -118,8 +123,42 @@ func (b *Builder) BuildOrDie(ctx context.Context) *Clients {
 	b.client.KubeClient = guestKubeClient
 	b.client.KubeInformers = v1helpers.NewKubeInformersForNamespaces(guestKubeClient, b.guestNamespaces...)
 
+	extractApplySpec := func(obj *unstructured.Unstructured, fieldManager string) (*applyoperatorv1.OperatorSpecApplyConfiguration, error) {
```
Comment (rhrmo) [inline pkg/clients/builder.go:126]: Please take extractApplySpec and extractApplyStatus outside of the function (so they are their own functions) and rename them to extractOperatorSpec and extractOperatorStatus

**Comment by dfajmon** [comment]
Closed in favor of #325

**Pull request closed without merging**